File: serialization/java/text/lib/src/main/java/com/microsoft/kiota/serialization/TextSerializationWriterFactory.java
package com.microsoft.kiota.serialization;

import java.util.Objects;

import javax.annotation.Nonnull;

import com.microsoft.kiota.serialization.SerializationWriter;
import com.microsoft.kiota.serialization.SerializationWriterFactory;

public class TextSerializationWriterFactory implements SerializationWriterFactory {
    public String getValidContentType() {
        return validContentType;
    }
    private static final String validContentType = "text/plain";
    @Override
    @Nonnull
    public SerializationWriter getSerializationWriter(@Nonnull final String contentType) {
        Objects.requireNonNull(contentType, "parameter contentType cannot be null");
        if(contentType.isEmpty()) {
            throw new NullPointerException("contentType cannot be empty");
        } else if (!contentType.equals(validContentType)) {
            throw new IllegalArgumentException("expected a " + validContentType + " content type");
        }
        return new TextSerializationWriter();
    }
}

File: serialization/php/text/src/TextSerializationWriterFactory.php
<?php
/**
 * Copyright (c) Microsoft Corporation.  All Rights Reserved.
 * Licensed under the MIT License.  See License in the project root
 * for license information.
 */


namespace Microsoft\Kiota\Serialization\Text;


use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriterFactory;

/**
 * Class TextSerializationWriterFactory
 * @package Microsoft\Kiota\Serialization\Text
 * @copyright 2022 Microsoft Corporation
 * @license https://opensource.org/licenses/MIT MIT License
 */
class TextSerializationWriterFactory implements SerializationWriterFactory
{

    /**
     * @inheritDoc
     */
    public function getSerializationWriter(string $contentType): SerializationWriter
    {
        if (strtolower($contentType) !== strtolower($this->getValidContentType())) {
            throw new \InvalidArgumentException("Expected content type to be {$this->getValidContentType()}");
        }
        return new TextSerializationWriter();
    }

    /**
     * @inheritDoc
     */
    public function getValidContentType(): string
    {
        return 'text/plain';
    }
}
